File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.py
def solution(number, limit, power):
    answer = []
    for n in range(1, number + 1):
        count = 0
        for i in range(1, int(n ** 0.5) + 1):
            if n % i == 0:
                count += 1
            
        if (n ** 0.5) == int(n ** 0.5):
            answer.append(count * 2 - 1)
        else:
            answer.append(count * 2)

    return sum([i if i <= limit else power for i in answer ])

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.js
function divisorCounter(num) {
    let result = new Set()
    if (num === 1) return 1
    
    for (let i = 1; i <= Math.sqrt(num); i++) {
        if (num % i === 0) result.add(i).add(num / i)
    }
    
    return result.size
} 
function solution(number, limit, power) {
    var answer = new Array(number).fill(0).map((_, i) => divisorCounter(i + 1)).reduce((acc, cur) => cur > limit ? acc += power : acc += cur,0)
    return answer;
}